validation.go

```diff
@@ -278,6 +278,10 @@ func (v *subSchema) validateRecursive(currentSubSchema *subSchema, currentNode i
 					return
 				}
 				value = cn
+			case primitive.Binary:
+				if !currentSubSchema.containsValidType(currentNode, result, context, TYPE_BINARY) {
+					return
+				}
 			}
```

Comment:
Do we need to set `value = cn` here as well? 

Also out of curiosity, do you know what it does?

Reply:
lol, i definitely need to do that. that is what actually sets the value of the node for validation. this also makes me think that I may need to add more validation. will look into that.

Reply:
do we have any limitations on the type of binary data that users can include in a schema? Or is any valid bson binary ok?

Reply:
I think any valid bson binary is okay. I think we'd just want to make sure it's binary data

Reply:
For synced collections, we're only going to allow BSON binary with subtype `0`, but we can do this check in the sync codebase instead of in `gojsonschema`, since any binary subtype is valid in non-sync scenarios